doc/Language/Proof-Search.org

```diff
@@ -0,0 +1,47 @@
+* Proof Searching of Juvix
+- Data Structures in Juvix should provide complimentary proofs that
+  are relevant to the users.
+** The Argument for Implicit Refinements Over Indexed Types For Structures
```

Comment:
This looks like a bit about what the standard library should provide? I think we might want a file specifically for the stdlib.

Reply:
We probably should have one, however I think this should stay here.